content/blog/2019/10/2019-10-04-vscode-caseStudy.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "JCasC JSONSchema Case-Study Using VSCode"
+opengraph:
+  image: /images/logos/JCasC/JCasC.png
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+=== JSON Schema and VSCode
+
+==== Configuration-as-code plugin
+The Configuration as Code plugin has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files. Writing such a file should be feasible without being a Jenkins expert, just translating into code a configuration process one is used to executing in the web UI.The plugin uses a schema to verify the files being applied to the Jenkins instance.
+
+With the new JSON Schema being enabled developers can now test their yaml file against it. The new schema checks for the configurators, their types and to a certain extent the nesting of the configurators. VSCode allows us to test out the schema right out of the box with some modifications.
+
+=== Steps to Enable the Schema Validation
+
+a) The first step includes installing the YAML plugin from redhat. Open up the Extension list in VSCode editor using `Ctrl + Shift + X`.
+
+b) Select and install the YAML plugin by Red Hat. Reload the editor if necessary.
+
+c) The next step includes inserting the JSON Schema inside the VSCode workspace. We would be including the schema in the `work` directory.
+
+d) Spin up your Jenkins server and navigate to `/configuration-as-code/jsonSchema`. 
+
+e) Copy and paste this schema into a new file called `json.schema`.
+
+f) In order to enable validation we need to include it in the workspace settings. In order to do that
+navigate to `File` and then `Preference` and then `Settings`. Inside settings search for `json` and inside `settings.json` include the following configuration.
+
+[source, json]
+
+---
+{
+"yaml.schemas": {
+        "schema.json": "jenkins.yml"
```

Comment:
the default configuration-as-code file is `jenkins.yaml` I'm not sure if both yml and yaml formats are picked up,
this will probably make it work for both but please test it:
```suggestion
        "schema.json": "jenkins.y[a]?ml"
```

Reply:
Yeah it works for me :)